File: Lecture-4/Code/AssociativeArray.cpp
#include <unordered_map> // or map

// mapping key: string to value: int
std::unordered_map<std::string, int> age;

if (age.count("Frank") > 0) // check for key
	age.erase("Frank"); // deletes a mapping

age.insert("Bob", 20);

// retrieves a value for key "Bob"
std::cout << "Bob is " << age["Bob"]
	<< " years old" << std::endl;

File: Lecture-4/Code/AssociativeArray.java
import java.util.HashMap; // or TreeMap

// mapping key: String to value: Integer
HashMap<String, Integer> age = new HashMap<>();

if (age.containsKey("Frank")) // check for key
	age.remove("Frank"); // deletes a mapping

age.put("Bob", 20); // insert entry

System.@{\textbf{\textit{\color{java_static}out}}}@.printf("Bob is %d years old",
	// retrieves a value for key "Bob"
	age.get("Bob"));